modules/Xsens_module/src/XsensRetargeting.cpp

```diff
@@ -0,0 +1,287 @@
+{
+    yarp::os::Bottle* desiredHumanJoints = m_wholeBodyHumanJointsPort.read(false);
+
+    if (desiredHumanJoints == NULL)
+    {
+        //        yError() << "[XsensRetargeting::getJointValues()] desiredWBJoints size: 0";
+        return true;
+    }
+
+    // in the msg thrift the first list is the joint names [get(0)], second the joint values
+    // [get(1)]
+    yarp::os::Value humanjointsValueS = desiredHumanJoints->get(1);
+    yarp::os::Bottle* tmpHumanNewJointValues = humanjointsValueS.asList();
+
+    yarp::sig::Vector newJointValues;
+    newJointValues.resize(m_actuatedDOFs, 0.0);
+
+    if (!m_firstIteration)
+    {
+        for (unsigned j = 0; j < m_actuatedDOFs; j++)
+        {
+            newJointValues(j) = tmpHumanNewJointValues->get(m_humanToRobotMap[j]).asDouble();
+            // check for the spikes in joint values
+            if (std::abs(newJointValues(j) - m_jointValues(j)) < m_jointDiffThreshold)
+            {
+                m_jointValues(j) = newJointValues(j);
+            } else
+            {
+                yWarning() << "spike in data: joint : " << j << " , " << m_robotJointsListNames[j]
+                           << " ; old data: " << m_jointValues(j)
+                           << " ; new data:" << newJointValues(j);
+            }
+        }
+    } else
+    {
+        yInfo() << "[XsensRetargeting::getJointValues] Xsens Retargeting Module is Running ...";
+        m_firstIteration = false;
+        for (unsigned j = 0; j < m_actuatedDOFs; j++)
+        {
+            newJointValues(j) = tmpHumanNewJointValues->get(m_humanToRobotMap[j]).asDouble();
+            m_jointValues(j) = newJointValues(j);
+        }
+        m_WBTrajectorySmoother->init(m_jointValues);
+    }
+
+    yInfo() << "joint [0]: " << m_robotJointsListNames[0] << " : " << newJointValues(0) << " , "
+            << m_jointValues(0);
+
+    //    m_tock =std::chrono::duration_cast< std::chrono::milliseconds
+    //    >(std::chrono::system_clock::now().time_since_epoch()); yDebug() << "------> rate: "<<
+    //    (m_tock-m_tick).count() << "ms";
+
+    //    m_tick =std::chrono::duration_cast< std::chrono::milliseconds
+    //    >(std::chrono::system_clock::now().time_since_epoch());
+
+    return true;
+}
+
+bool XsensRetargeting::getSmoothedJointValues(yarp::sig::Vector& smoothedJointValues)
+{
+    m_WBTrajectorySmoother->computeNextValues(m_jointValues);
+    smoothedJointValues = m_WBTrajectorySmoother->getPos();
+
+    return true;
+}
+
+double XsensRetargeting::getPeriod()
+{
+    return m_dT;
+}
+
+bool XsensRetargeting::updateModule()
+{
+
+    getJointValues();
+
+    if (m_wholeBodyHumanJointsPort.isClosed())
+    {
+        yError() << "[XsensRetargeting::updateModule] m_wholeBodyHumanJointsPort port is closed";
+        return false;
+    }
+
+    // DEL: delete this check here! it was initially applied to check if the joint values at the
+    // beginning does not go to 0.0 .
+    if (!m_firstIteration)
```

Comment:
@GiulioRomualdi  I think It is better I remove this check here, as we resolve the problem of zero initialization of the `minJerkTrajectory`.

Reply:
Actually, I do not understand the meaning of the following code
```cpp
        double temp_jointsNorm = 0.0, tmp_normThreshold = 0.001;
        for (unsigned i = 0; i < m_jointValues.size(); i++)
        {
            temp_jointsNorm += (m_jointValues(i) * m_jointValues(i));
        }
        temp_jointsNorm = std::sqrt(temp_jointsNorm);

        if (temp_jointsNorm > tmp_normThreshold)
        {
            yarp::sig::Vector& refValues = m_wholeBodyHumanSmoothedJointsPort.prepare();
            getSmoothedJointValues(refValues);
            m_wholeBodyHumanSmoothedJointsPort.write();
        } else
        {
            yWarning() << "The norm of joint values are less than the given threshold ( "
                       << "joint  norm: " << temp_jointsNorm
                       << " , threshold: " << tmp_normThreshold
                       << " ), is not passed to Controller ";
        }
``` 
why are you checking the norm of the desired joint values?

Reply:
because, before there was a problem about initialization of the minJerkTrajectory, so at the beginning I added this check. Since the problem is resolved, I will delete this check.